### PR TITLE
Attribute Module Spec

### PR DESCRIPTION
**Pull request by Taztingo**
## Description

Adds the standard spec to the attribute module.

closes: #91

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: x/attribute/spec/01_state.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# State
+The attribute module inserts all attributes into a basic state store.
+
+# Attribute KV-Store
```
Comment (dwedul-figure) [inline x/attribute/spec/01_state.md:4]: This should be level 2 and the ones below it should be level 3.
Typically there's only ever one level 1 header in these files, and that's the one at the very top defining the whole document.

File: x/attribute/spec/02_messages.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
+# Messages
+
+In this section we describe the processing of the staking messages and the corresponding updates to the state.
+
```
Comment (dwedul-figure) [inline x/attribute/spec/02_messages.md:4]: Could use a `<!-- TOC -->` here.

**Review by dwedul-figure — COMMENTED** [review]
For the TOC stuff, A comment of `<!-- TOC -->` is the same as `<!-- TOC 2 3 -->` which stands for "Make a table of contents of headings of levels 2 through 3. That'd be sufficient for the two files that could use a TOC. It's usually nice to have an empty line before the comment too.

After you add the comments, run `make update-tocs`. That'll create the TOC content as needed. 

Alternatively, Instead of manually placing the `<!-- TOC -->` comment and using the make target, you could run `scripts/update-toc.sh --force x/attribute/spec/01_state.md x/attribute/spec/02_messages.md`. That'll create and place the TOC in the standard place - above the first level 2 item.

The make target only generates TOCs for files that already have a TOC comment. It uses that same bash script. But to add a TOC to a file that doesn't have a TOC comment, it needs the --force flag.

File: x/attribute/spec/01_state.md (hunk 1 above)
Comment (Taztingo) [reply x/attribute/spec/01_state.md]: Added

File: x/attribute/spec/01_state.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
+# State
+The attribute module inserts all attributes into a basic state store.
+
```
[the remark this answers is not in the document]
Comment (Taztingo) [reply x/attribute/spec/01_state.md:3]: Added

File: x/attribute/spec/02_messages.md (hunk 2 above)
Comment (Taztingo) [reply x/attribute/spec/02_messages.md:4]: Added

**Review by dwedul-figure — APPROVED** [review]

**Review by egaxhaj-figure — APPROVED** [review]
LGTM

**Pull request merged**